File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        d = {}
        
        for i in range(len(nums)):
            d[nums[i]] = i
        
        for i in range(len(nums)):
            if target-nums[i] in d and i != d[target-nums[i]]:
                return [i, d[target-nums[i]]]
        

File: 0001-two-sum/0001-two-sum.go
type s struct {
    p bool
    i int
}

func twoSum(nums []int, target int) []int {
    
    d := make(map[int]s)
    
    for i, e := range nums {
        n := target - e
        if d[n].p == true {
            return []int{d[n].i, i}
        }
        
        x := new(s)
        x.p = true
        x.i = i
        
        d[e] = *x
    }
    
    return []int{-1,-1}
}